### PR TITLE
Fix RDF and TTreeReader part of ROOT-10702, add tests

### PR DESCRIPTION
**Pull request by eguiraud**
For some ROOT files, RDF and TTreeReader failed to retrieve branch `"i.x"`. Fixed by using `FindBranch` instead of `GetBranch` in the relevant routines.

~~Unfortunately, this seems to be a noticeable performance pessimization, but I'm out of ideas.~~ Fixed, see Axel's suggestion.

Relevant tests are added, but will fail as long as `GetBranch("x")` returns `i.x` instead of the top-level `x` branch.

**About test failures**

* the failures in `dataframe_snapshot`, `testTTreeRegressions` and `treetreeplayertestUnit` are those I mention above, and are expected
* if I understand correctly the failures in `roottest_root_tree_selectorreader_make` and `roottest_root_tree_reader_make` show that, surprisingly, `FindBranch` is not strictly more powerful than `GetBranch`: there are nested branches that could be found before and cannot be found now (EDIT: fixed by Axel's suggestion to default to `GetBranch` and go to `FindBranch` only if it returns a nullptr)
* I don't understand the failures in `roottest_root_tree_entrylist_make` and `roottest_root_tree_chain_make`, my PR does not touch anything related to TTree/TChain/TEntryList

### CODE REVIEW

File: tree/dataframe/src/RLoopManager.cxx [hunk 1]
```diff
@@ -97,7 +97,7 @@ static void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnName
 
       ExploreBranch(t, bNamesReg, bNames, subBranch, newPrefix, friendName);
 
-      if (t.GetBranch(fullName.c_str())) {
```
Comment (Axel-Naumann) [inline tree/dataframe/src/RLoopManager.cxx:100]: Can we do `GetBranch() || FindBranch()` to reduce the perf cost?

Comment (eguiraud) [reply tree/dataframe/src/RLoopManager.cxx:100]: yes! good idea, that would even fix the problems due to `FindBranch` not finding what `GetBranch` can find or `FindBranch` returning a different thing than `GetBranch`.

Comment (eguiraud) [reply tree/dataframe/src/RLoopManager.cxx:100]: done

**Comment by pcanal** [comment]
>  surprisingly, FindBranch is not strictly more powerful than GetBranch: there are nested branches that could be found before and cannot be found now

I suppose that technically that is a bug in FindBranch ...

**Comment by pcanal** [comment]
> I don't understand the failures in roottest_root_tree_entrylist_make and roottest_root_tree_chain_make, my PR does not touch anything related to TTree/TChain/TEntryList

They appears to be bug in cmake/ctest/jenkins.  Something that can't possibly not be there is not found (or vice et versa, asking for something that is not supposed to be there).

**Comment by eguiraud** [comment]
> Likely a cmake race condition.

mmh i love those :smile:

File: tree/dataframe/test/dataframe_snapshot.cxx [hunk 2]
```diff
@@ -705,6 +706,19 @@ TEST(RDFSnapshotMore, TClonesArray)
    ReadWriteTClonesArray();
 }
 
+// ROOT-10702
+TEST(RDFSnapshotMore, CompositeTypeWithNameClash)
+{
+	const auto fname = "snap_compositetypewithnameclash.root";
+	gInterpreter->Declare("struct Int { int x; };");
+   ROOT::RDataFrame df(3);
+   auto snap_df = df.Define("i", "Int{-1};").Define("x", [] { return 1; }).Snapshot("t", fname);
```
Comment (hageboeck) [inline tree/dataframe/test/dataframe_snapshot.cxx:715]: Indentation got messed up here.
```suggestion
	ROOT::RDataFrame df(3);
	auto snap_df = df.Define("i", "Int{-1};").Define("x", [] { return 1; }).Snapshot("t", fname);
```

Comment (eguiraud) [reply tree/dataframe/test/dataframe_snapshot.cxx:715]: ah good catch, thanks

**Pull request merged**